clippy_lints/src/manual_checked_sub.rs

```diff
@@ -0,0 +1,239 @@
+                    condition_rhs: rhs,
+                }
+                .visit_expr(body_block);
+            }
+
+            if let BinOpKind::Gt = op.node {
+                if let ExprKind::Lit(lit) = &rhs.kind
+                    && let LitKind::Int(Pu128(0), _) = lit.node
+                {
+                    SubExprVisitor {
+                        cx,
+                        if_expr: expr,
+                        if_body_block: body_block,
+                        else_block,
+                        applicability,
+                        condition_lhs: lhs,
+                        condition_rhs: rhs,
+                    }
+                    .visit_expr(body_block);
+                }
+            }
+        }
+    }
+}
+
+struct SubExprVisitor<'a, 'tcx> {
+    cx: &'a LateContext<'tcx>,
+    if_expr: &'tcx Expr<'tcx>,
+    if_body_block: &'tcx Expr<'tcx>,
+    else_block: Option<&'tcx Expr<'tcx>>,
+    applicability: Applicability,
+    condition_lhs: &'tcx Expr<'tcx>,
+    condition_rhs: &'tcx Expr<'tcx>,
+}
+
+impl<'tcx> Visitor<'tcx> for SubExprVisitor<'_, 'tcx> {
+    fn visit_expr(&mut self, expr: &'tcx Expr<'_>) {
+        if let ExprKind::Binary(op, sub_lhs, sub_rhs) = expr.kind
+            && let BinOpKind::Sub = op.node
+        {
+            if let ExprKind::Lit(lit) = self.condition_rhs.kind
+                && let LitKind::Int(Pu128(0), _) = lit.node
+                && (is_referencing_same_variable(sub_lhs, self.condition_lhs)
+                    || are_literals_equal(sub_lhs, self.condition_lhs))
+            {
+                self.build_suggession(expr);
+            } else if (is_referencing_same_variable(sub_lhs, self.condition_lhs)
+                || are_literals_equal(sub_lhs, self.condition_lhs))
+                && (is_referencing_same_variable(sub_rhs, self.condition_rhs)
+                    || are_literals_equal(sub_rhs, self.condition_rhs))
+            {
+                self.build_suggession(expr);
+            }
+        }
+
+        walk_expr(self, expr);
+    }
+}
+
+impl<'tcx> SubExprVisitor<'_, 'tcx> {
+    fn build_suggession(&mut self, expr: &Expr<'tcx>) {
+        let body_snippet = snippet_with_applicability(self.cx, self.if_body_block.span, "..", &mut self.applicability);
+
+        let binary_expr_snippet = snippet_with_applicability(self.cx, expr.span, "..", &mut self.applicability);
+        let updated_usage_context_snippet = body_snippet.as_ref().replace(binary_expr_snippet.as_ref(), "result");
+        // let else_snippet = snippet_with_applicability(self.cx, self.if_else_block.span, "..", &mut
+        // self.applicability);
+
+        let lhs_snippet = snippet_with_applicability(self.cx, self.condition_lhs.span, "..", &mut self.applicability);
+        let rhs_snippet = snippet_with_applicability(self.cx, self.condition_rhs.span, "..", &mut self.applicability);
+
+        let lhs_needs_parens = matches!(self.condition_lhs.kind, ExprKind::Cast(..));
```

Comment:
Maybe you should use the `clippy_utils::sugg::Sugg` interface, which lets you deal with parentheses in a more "standardized" way.